include/pagmo/algorithms/gaco.hpp

```diff
@@ -0,0 +1,988 @@
+                        "The focus parameter must be >=0  while a value of " + std::to_string(focus) + " was detected");
+        }
+
+        if (epsilon >= 1. || epsilon < 0.) {
+            pagmo_throw(std::invalid_argument, "The Pareto precision parameter must be in [0, 1[, while a value of "
+                                                   + std::to_string(epsilon) + " was detected");
+        }
+        if ((threshold < 1 || threshold > gen) && gen != 0 && memory == false) {
+            pagmo_throw(std::invalid_argument,
+                        "If memory is inactive, the threshold parameter must be either in [1,m_gen] while a value of "
+                            + std::to_string(threshold) + " was detected");
+        }
+        if (threshold < 1 && gen != 0 && memory == true) {
+            pagmo_throw(std::invalid_argument,
+                        "If memory is active, the threshold parameter must be >=1 while a value of "
+                            + std::to_string(threshold) + " was detected");
+        }
+    }
+
+    /// Algorithm evolve method (juice implementation of the algorithm)
+    /**
+     *
+     * Evolves the population for the requested number of generations.
+     *
+     * @param pop population to be evolved
+     * @return evolved population
+     * @throw std::invalid_argument if pop.get_problem() is stochastic.
+     */
+
+    population evolve(population pop) const
+    {
+        // If the memory is active, we increase the counter:
+        if (m_memory == true) {
+            ++m_counter;
+        }
+
+        // We store some useful variables:
+        const auto &prob = pop.get_problem(); // This is a const reference, so using set_seed for example will not be
+        // allowed
+        auto n_x = prob.get_nx(); // This getter does not return a const reference but a copy of the number of
+        // all the variables
+        auto pop_size = pop.size(); // Population size
+        unsigned count_screen = 1u; // regulates the screen output
+
+        // Note that the number of equality and inequality constraints has to be set up manually in the problem
+        // definition, otherwise PaGMO assumes that there aren't any.
+        auto n_obj = prob.get_nobj();
+        auto n_ec = prob.get_nec();
+        auto n_ic = prob.get_nic();
+        auto n_f = prob.get_nf(); // n_f=prob.get_nobj()+prob.get_nec()+prob.get_nic()
+
+        // Other useful variables are stored:
+        std::vector<vector_double> sol_archive(m_ker, vector_double(1 + n_x + n_f, 1));
+        vector_double omega(m_ker);
+        vector_double prob_cumulative(m_ker);
+        std::uniform_real_distribution<> dist(0, 1);
+        std::normal_distribution<double> gauss{0., 1.};
+
+        // PREAMBLE-------------------------------------------------------------------------------------------------
+        // We start by checking that the problem is suitable for this
+        // particular algorithm.
+
+        if (!pop_size) {
```

Comment:
For an empty population, there's no need to throw. Just return the original (empty) population immediately.